src/core/Akka/Util/Extensions/TaskExtensions.cs

```diff
@@ -0,0 +1,55 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="TaskExtensions.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2022 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2022 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Akka.Util.Extensions
+{
+    internal static class TaskExtensions
+    {
+#if NETSTANDARD
+        public static async Task WaitAsync(this Task task, TimeSpan timeout)
```

Comment:
Separate issue that can be resolved at a later time: wonder if it's possible to reduce allocations around this via `ValueTask` or some of the other concepts + value types that were introduced in the .NET 6 implementation: https://andrewlock.net/a-deep-dive-into-the-new-task-waitasync-api-in-dotnet-6/

Not huge issue worth spending a lot of cycles on right now unless they're easy wins.

Reply:
I'll include a improved version, based on the .NET6 implementation and with `CancellationToken` support to properly cancel the task.